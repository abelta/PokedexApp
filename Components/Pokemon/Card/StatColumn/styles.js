import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  statColumn: {
    flex: 1,
    flexDirection: 'column',
    height: 120,
    marginLeft: 5,
    marginRight: 5,
  },
  filler: {
  },
  label: {
    backgroundColor: 'lightgrey',
    height: 20,
    fontSize: 12,
    textAlign: 'center',
  },
});

export default styles;
