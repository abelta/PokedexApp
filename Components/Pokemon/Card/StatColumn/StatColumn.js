import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import styles from './styles';

const statColor = (label) => {
  switch (label) {
    case 'speed': return 'pink';
    case 'special-defense': return 'lightgreen';
    case 'special-attack': return 'lightblue';
    case 'defense': return 'yellow';
    case 'attack': return 'orange';
    case 'hp': return 'red';
    default: return 'black';
  }
};

const statLabel = (label) => {
  switch (label) {
    case 'speed': return 'Speed';
    case 'special-defense': return 'Sp. Def';
    case 'special-attack': return 'Sp. Atk';
    case 'defense': return 'Def';
    case 'attack': return 'Atk';
    case 'hp': return 'HP';
    default: return 'black';
  }
};

const StatColumn = ({ label, value }) => (
  <View key={label} style={styles.statColumn}>
    <View style={[styles.filler, { height: value, backgroundColor: statColor(label) }]} />
    <Text style={styles.label}>{statLabel(label)}</Text>
  </View>
);

StatColumn.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
};

export default StatColumn;
