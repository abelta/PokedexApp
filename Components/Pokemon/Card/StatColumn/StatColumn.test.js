import React from 'react';
import renderer from 'react-test-renderer';
import StatColumn from '.';

describe('StatColumn', () => {
  it('renders properly', () => {
    const tree = renderer.create(<StatColumn label="label" value={50} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
