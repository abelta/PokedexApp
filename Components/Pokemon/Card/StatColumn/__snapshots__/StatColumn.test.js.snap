// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatColumn renders properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "height": 120,
      "marginLeft": 5,
      "marginRight": 5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "bottom": 20,
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "backgroundColor": "black",
          "height": 50,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "lightgrey",
        "bottom": 0,
        "fontSize": 12,
        "height": 20,
        "position": "absolute",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    label
  </Text>
</View>
`;
